unit-tests/live/frames/fps_helper.py

```diff
@@ -0,0 +1,169 @@
+
+from rspy import test, log
+import time
+
+# global variable used to count on all the sensors simultaneously
+count_frames = False
+# tests parameters
+TIME_FOR_STEADY_STATE = 3
+TIME_TO_COUNT_FRAMES = 5
+
+
+##########################################
+# ---------- Helper Functions ---------- #
+##########################################
+def check_fps_pair(measured_fps, expected_fps):
+    delta_Hz = expected_fps * 0.05  # Validation KPI is 5%
+    return (measured_fps <= (expected_fps + delta_Hz) and measured_fps >= (expected_fps - delta_Hz))
+
+
+def get_expected_fps_dict(sensor_profiles_dict):
+    """
+    Returns a dictionary between each stream name and its expected fps
+    """
+    expected_fps_dict = {profile.stream_name(): profile.fps()
+                         for profiles in sensor_profiles_dict.values()
+                         for profile in profiles}
+    return expected_fps_dict
+
+
+def get_dict_for_streams(sensor_profiles_arr, streams_to_test):
+    sensor_profiles_dict = {}
+    for stream_name in streams_to_test:
+        for sensor, profile in sensor_profiles_arr:
+            if stream_name in profile.stream_name():
+                sensor_profiles_dict[sensor] = sensor_profiles_dict.get(sensor, []) + [profile]
+    return sensor_profiles_dict
+
+
+def get_time_est_string(profiles_array):
+    global TIME_TO_COUNT_FRAMES, TIME_FOR_STEADY_STATE
+    time_per_test = TIME_TO_COUNT_FRAMES + TIME_FOR_STEADY_STATE
+    time_est_string = (f"Estimated time for test: {len(profiles_array) * time_per_test} secs "
+                       f"({len(profiles_array)} tests * {time_per_test} secs per test)")
+    return time_est_string
+
+
+def get_tested_profiles_string(sensor_profiles_dict):
+    tested_profile_string = ""
+    for sensor, profiles in sensor_profiles_dict.items():
+        for profile in profiles:
+            tested_profile_string += f"{sensor.name} / {profile.stream_name()} + "
+
+    # Remove the last " + "
+    tested_profile_string = tested_profile_string[:-3]
+    return tested_profile_string
+
+
+#############################################
+# ---------- Core Test Functions ---------- #
+#############################################
+def check_fps_dict(measured_fps, expected_fps):
+    all_fps_ok = True
+    for profile_name in expected_fps:
+        res = check_fps_pair(measured_fps[profile_name], expected_fps[profile_name])
+        if not res:
+            all_fps_ok = False
+        log.d(f"Expected {expected_fps[profile_name]} fps, received {measured_fps[profile_name]} fps in profile"
+              f" {profile_name}"
+              f" {'(Pass)' if res else '(Fail)'}")
+    return all_fps_ok
+
+
+def generate_callbacks(sensor_profiles_dict, profile_name_fps_dict):
+    """
+    Creates callable functions for each sensor to be triggered when a new frame arrives
+    Used to count frames received for measuring fps
+    """
+    def on_frame_received(frame):
```

Comment:
define this function outside of generate_callbacks and then maybe remove this function